view/adminhtml/templates/tracking.phtml

```diff
@@ -1,4 +1,11 @@
+<?php
+/** @var \Algolia\AlgoliaSearch\ViewModel\Adminhtml\Common $viewModel */
+$view = $this->getViewModel();
+$applicationId = $view->getApplicationId();
+?>
+
 <script>
+var applicationId = <?php echo json_encode($applicationId); ?>;
```

Comment:
based on Magento documentation and XSS prevention for single string (not object) Magento recommends to use escapeJS (ref. https://devdocs.magento.com/guides/v2.3/extension-dev-guide/xss-protection.html)